src/Framework/AspNetCoreAnalyzers/src/Analyzers/Resources.resx

```diff
@@ -315,4 +315,10 @@
   <data name="Analyzer_UseAddAuthorizationBuilder_Title" xml:space="preserve">
     <value>Use AddAuthorizationBuilder</value>
   </data>
+  <data name="Analyzer_AuthorizeAttributeOverridden_Message" xml:space="preserve">
+    <value>This [Authorize] attribute is overridden by an [AllowAnonymous] attribute from farther away on '{0}'</value>
```

Comment:
I think the title below is good, but for the message I think people who don't know what is going on will want some more detail.

More information:
* When authorization is run, AllowAnonymous takes precedence over Authorize. If both are present, then authorization doesn't run.
* MVC actions inherit (is that the right word? You should look at routing docs to see what it says) metadata from the controller. And controllers inherit metadata from base controls.
* Investigate your controllers and actions and decide whether to remove the [Authorize] attribute or move where the [AllowAnonymous] attribute is placed.

That detail could be in the message. Or there could be an aka.ms link here. Or it could be in the doc for the analyzer warning on learn.microsoft.com

Reply:
> When authorization is run, AllowAnonymous takes precedence over Authorize. If both are present, then authorization doesn't run.

Does this need a bit of refinement? It sounds we are saying the filter doesn't run, but that's not truly the case. The filter still runs partially, it just doesn't call `policy.AuthorizeAsync`. So maybe we can clarify a bit saying that in this case the filter will run the authentication phase but not the authorization phase. This is relevant because the authentication phase can change the principal.

See https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.Core/src/Authorization/AuthorizeFilter.cs#L161-L183 for details.

At least `AuthenticateAsync` happens, which sets the principal on to the context. See https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authorization/Policy/src/PolicyEvaluator.cs#L56